core/api_token_api.php

```diff
@@ -117,11 +124,15 @@ function api_token_validate( $p_username, $p_token ) {
 		return false;
 	}
 
-	$t_encrypted_token = api_token_hash( $p_token );
-
+	if( API_TOKEN_GENERIC == $p_kind ) {
+		$t_encrypted_token = api_token_hash( $p_token );
+	} else {
+		$t_encrypted_token = api_token_hash( $p_kind . $p_token );
```

Comment:
Why do we need a different encryption type?  I would drop this.  Each token should be long and unique on its own.  You can imagine a token type being modified in the future.